spec/helpers/sticky_footer_visibility_spec.rb

```diff
@@ -0,0 +1,24 @@
+RSpec.describe StickyNewsletterVisibility, type: :helper do
+  let(:request) { double('request')}
+
+  describe 'home page' do
+    %w(en cy).each do |locale|
+      it 'excludes in any language' do
+        allow(request).to receive(:url) { "http://example.com/#{locale}" }
+        allow(request).to receive(:base_url) { 'http://example.com' }
+        expect(display_sticky_newsletter_on_this_slug?).to be false
+      end
+    end
+  end
+
+  describe 'category slugs' do
+    %w(en cy).each do |locale|
+      it 'excludes in any language' do
+        allow(request).to receive(:url) { "http://example.com/#{locale}/categories/insurance" }
+        allow(request).to receive(:base_url) { 'http://example.com' }
+
+        expect(display_sticky_newsletter_on_this_slug?).to be false
+      end
+    end
+  end
+end
```

Comment:
All this tests can be moved the allow expectations to a before clause and being divide using lets. Or instead of using before you can add the expectation to the double and using lets too.